src/app/blog/iroh-0-33-0-browsers-and-discovery-and-0-RTT-oh-my/page.mdx

```diff
@@ -0,0 +1,111 @@
+        - `iroh::dns::node_info::{to_z32, from_z32}`are removed. Use the methods on `iroh::dns::node_info::NodeIdExt` trait instead.
+        - `iroh::dns::ResolverExt` is removed. Use the methods on `iroh::dns::DnsResolver` instead.
+        - `iroh::discovery::Discovery::publish` now takes `data: &NodeData` as its single argument. `iroh::discovery::NodeData` is a re-export of `iroh_relay::dns::node_info::NodeData`, and contains relay URL and direct addresses. See docs for `NodeData` for details.
+        - `iroh::Endpoint::connect_with` was removed, use `iroh::Endpoint::connect_with_opts` instead
+    - changed
+        - `iroh::dns::DnsResolver` used to be a type alias and now is a reexport of `iroh_relay::dns::DnsResolver` struct
+        - `iroh::dns::node_info` module is now a reexport of `iroh_relay::dns::node_info`
+        - `iroh::discovery::dns::{N0_DNS_NODE_ORIGIN_PROD, N0_DNS_NODE_ORIGIN_STAGING}` are now reexports of `iroh_relay::dns::{N0_DNS_NODE_ORIGIN_PROD, N0_DNS_NODE_ORIGIN_STAGING}`
+        - The methods in `iroh::dns::DnsResolver` now take an `impl ToString` instead of `impl hickory_proto::rr::domain::IntoName` for their `host` argument
+        - `iroh::discovery::DiscoveryItem` no longer has any public fields. There are now getters for the contained data, and constructors for createing a `DiscoveryItem` from a `NodeInfo`.
+        - `iroh_relay::dns::node_info::NodeInfo` is changed.
+            - `NodeInfo::new` now has a single `NodeId` argument. Use `NodeInfo::with_direct_addresses` and `NodeInfo::with_relay_url` to set addresses and relay URL. Alternatively, use `NodeInfo::from_parts` and pass a `NodeData` struct.
+            - `NodeInfo` now has two public fields `node_id: NodeId` and `data: NodeData`, and setter and getter methods for the relay URL and direct addresses.
+        - `iroh::discovery::pkarr::PkarrPublisher::update_addr_info` now takes a `NodeData` argument
+        - `iroh::endpoint::Connection::into_0rtt` now returns `iroh::endpoint::ZeroRttAccepted`, instead of `iroh_quinn::ZeroRttAccepted`
+
+### But wait, there's more!
+
+Many bugs were squashed, and smaller features were added. For all those details, check out the full changelog: [https://github.com/n0-computer/iroh/releases/tag/v0.33.0](https://github.com/n0-computer/iroh/releases/tag/v0.29.0).
+
+If you want to know what is coming up, check out the [v0.34.0 milestone](https://github.com/n0-computer/iroh/milestone/40), and if you have any wishes, let us know about the [issues](https://github.com/n0-computer/iroh/issues)! If you need help using iroh or just want to chat, please join us on [discord](https://discord.com/invite/DpmJgtU7cW)! And to keep up with all things iroh, check out our [Twitter](https://x.com/iroh_n0).
```

Comment:
The link for "0.34 milestone" links to the 0.33 one. Is that intentional, will it get renamed?